torch/csrc/jit/ir.h

```diff
@@ -736,7 +736,9 @@ struct TORCH_API Node {
     return getAttr<TensorsAttr>(name);
   }
 
-private:
+  Block* findCommonAncestorBlockWith(Node* n);
```

Comment:
There are also a number of places where we do stuff like this, it could be nice to clean them up by using this API. Topological moves come to mind

Reply:
And…DCE? maybe? I don't remember

Reply:
This function could be refactored to using this API: https://github.com/pytorch/pytorch/blob/4bfe2f0900a602017a31c0aaa8f00750412a2d83/torch/csrc/jit/ir.cpp#L1176

Reply:
Hmm, I tried using this API for that `isBeforeOrAfter` but `isBeforeOrAfter` requires having pointers to the ancestors Nodes that reside in the common ancestor block

Reply:
Sorry i meant to say `isBeforeOrAfter` could be refactored using this function. but actually i take it back